internal/wasm/memory_test.go

```diff
@@ -733,6 +733,13 @@ func TestMemoryInstance_Write(t *testing.T) {
 
 	ok = mem.Write(9, buf)
 	require.False(t, ok)
+
+	// Test overflow
+	huge := uint64(math.MaxUint32 + 1 + 4)
```

Comment:
I love I didn't have to ask for this test

Reply:
Let me see if it goes through in 32-bit.
Does CI check this?
I can pull it on Linux and trying running there.

Reply:
we don't run tests on 32-bit platforms at the moment, only compilation tests.

Reply:
Pass on `GOARCH=386`, fail with `race`.
I'll isolate into its own test, catch the panic and skip the test.